apps/www/_blog/2024-07-01-supabase-realtime-broadcast-and-presence-authorization.mdx

```diff
@@ -0,0 +1,186 @@
+This release introduces authorization for Broadcast and Presence using Row Level Security policies:
+
+```sql
+CREATE POLICY "authenticated user can listen to all messages"
+ON "realtime"."messages"
+AS PERMISSIVE FOR SELECT
+TO authenticated
+USING ( true );
+```
+
+This follows a similar approach to [Supabase Storage](https://supabase.com/docs/guides/storage/security/access-control) where we create a new table in the `realtime` schema that represents your [Channels](https://supabase.com/docs/guides/realtime/concepts#channels) as objects so you can write policies where SELECT and INSERT will determines what your users can access and use from Realtime.
+
+## How it works without Authorization
+
+When you want to connect to a Realtime Channel, you do something close to the following code:
+
+```tsx
+// With authenticated user token
+client.realtime.setAuth(token)
+
+client.channel('channel').subscribe((status: string, err: any) => {
+  status === 'SUBSCRIBED' && console.log('Connected')
+})
+```
+
+This will enable you to send messages but they will all be public and every user would be able to intercept them if they had your anon token.
+
+To convert this into a Realtime Channel that checks RLS policies requires only two steps:
+
+- [Create RLS policies](#create-rls-policies)
+- [Enabling Authorization on Channel subscription](#enabling-authorization-on-channel-subscription)
+
+### Create RLS policies
+
+Any RLS policy will work. Reference our [Github Discussion](github.com/orgs/supabase/discussions/22484) for more complex use cases.
+
+With this example we'll keep it simple and allow authenticated users to connect to a channel, send and receive broadcast messages and track presences.
+
+```sql
+CREATE POLICY "authenticated user listen to all"
+ON "realtime"."messages"
+AS PERMISSIVE FOR SELECT
+TO authenticated
+USING ( true );
+
+CREATE POLICY "authenticated user write to all"
+ON "realtime"."messages"
+AS PERMISSIVE FOR INSERT
+TO authenticated
+WITH CHECK ( true );
+```
+
+We also have a new database function called`realtime.topic()` which enables you to write Policies that take into consideration the name of the Channel trying to run the RLS Policy.
+
+```sql
+create policy "authenticated users can only read from 'locked' topic"
+on "realtime"."messages"
+as PERMISSIVE for SELECT
+to authenticated
+using ( realtime.topic() = 'locked' );
+```
+
+## Enabling Authorization on Channel subscription
+
+Now that you have your RLS policies set you will need to define on your Channel subscription that you intend to use a private channel. To achieve it we've introduced a new configuration parameter `private` to signal to Realtime servers that you want to check authorization on the channel.
+
+If you try to subscribe with an unauthorized user you will get a new error message informing the user that they do not have permission to access the topic.
+
+```tsx
+// With anon user
+supabase
+    .realtime
+    .channel("locked", { config: { private: true })
```

Comment:
```suggestion
    .channel("locked", { config: { private: true } })
```

Out of curiosity, what happens if you try to join a private channel without passing `private: true`? Assuming you get a similar error? 

Not super important, but it was the first question that popped to mind while reading this, so might be worth nothing briefly.

Reply:
Not possible as this will be part of the websocket initialising payload and that will determine the topic to which they are connected internally.

e.g.
private: false - user will be connected to a topic realtime:topic
private: true - user will be connected to a topic realtime-private:topic